examples/prom/nonlinear_elasticity_global_rom.cpp

```diff
@@ -0,0 +1,1749 @@
+﻿//               libROM MFEM Example: parametric ROM for nonlinear elasticity problem (adapted from ex10p.cpp)
+//
+// Compile with: make nonlinear_elasticity_global_rom
+//
+// Description:  This examples solves a time dependent nonlinear elasticity
+//               problem of the form dv/dt = H(x) + S v, dx/dt = v, where H is a
+//               hyperelastic model and S is a viscosity operator of Laplacian
+//               type. The geometry of the domain is assumed to be as follows:
+//
+//                                 +---------------------+
+//                    boundary --->|                     |
+//                    attribute 1  |                     |
+//                    (fixed)      +---------------------+
+//
+//               The example demonstrates the use of hyper reduction to solve a
+//               nonlinear elasticity problem. Time integration is done with various
+//               explicit time integrator solvers. The basis for the velocity field
+//               is either constructed using a separate velocity basis or using the
+//               displacement basis. It is possible to set the initial condition in
+//               terms of either velocity or deformation. The velocity initial condition
+//               works better when both velocity and displacement bases are used. The
+//               deformation initial condition is better when only the displacement
+//               basis is used. The input flag -sc controls the scaling of the initial
+//               condition applied. This is what parameterizes the ROM. If the scaling
+//               factor is within the range +-10%, the results are generally accurate.
+
+// =================================================================================
+//
+// Sample runs and results for parametric ROM using displacement basis, velocity basis
+// and nonlinear term basis, with velocity initial condition:
+//
+// Offline phase:
+//      ./nonlinear_elasticity_global_rom --offline -dt 0.01 -tf 5.0 -s 14 -vs 100 -sc 0.90 -id 0
+//
+//      ./nonlinear_elasticity_global_rom --offline -dt 0.01 -tf 5.0 -s 14 -vs 100 -sc 1.10 -id 1
+//
+// Merge phase:
+//      ./nonlinear_elasticity_global_rom --merge -ns 2 -dt 0.01 -tf 5.0
+//
+// Create FOM comparison data:
+//      ./nonlinear_elasticity_global_rom --offline -dt 0.01 -tf 5.0 -s 14 -vs 100 -sc 1.00 -id 2
+//
+// Online phase with full sampling:
+//      ./nonlinear_elasticity_global_rom --online -dt 0.01 -tf 5.0 -s 14 -vs 100 -hyp -rvdim 40 -rxdim 10 -hdim 71 -nsr 1170 -sc 1.00
+// Output message:
+//      Elapsed time for time integration loop 14.0381
+//      Relative error of ROM position (x) at t_final: 5 is 2.97882e-06
+//      Relative error of ROM velocity (v) at t_final: 5 is 0.000679751
+//
+// Online phase with strong hyper reduction:
```

Comment:
This is indeed strong hyperreduction. With `nsr 100` the speedup is still good, without about 3% error. Maybe a brief comment about that would be helpful. Otherwise, the reader has to experiment to find this.

Reply:
I have included -nsr 100 case below. I have also did astyle. @dylan-copeland if you can review one more time, that would be wonderful.

Reply:
Hi @chldkdtn @dylan-copeland Sorry I've been unresponsive for a couple of days. A lot of school work unfortunately. Thank you, @chldkdtn for adding the example and doing astyle. Is there anything more I can help out with?